dynawo/sources/Models/CPP/Events/ModelVoltageSetPointChange/DYNModelVoltageSetPointChange.h

```diff
@@ -0,0 +1,288 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools
+// for power systems.
+//
+
+/**
+ * @file  DYNVoltageSetPointChange.h
+ *
+ * @brief
+ *
+ */
+#ifndef MODELS_CPP_EVENTS_MODELVOLTAGESETPOINTCHANGE_DYNMODELVOLTAGESETPOINTCHANGE_H_
+#define MODELS_CPP_EVENTS_MODELVOLTAGESETPOINTCHANGE_DYNMODELVOLTAGESETPOINTCHANGE_H_
+
+#include "DYNModelCPPImpl.h"
+#include "DYNSubModelFactory.h"
+
+namespace DYN {
+
+/**
+ * @brief ModelVoltageSetPointChangeFactory Model factory
+ *
+ * Implementation of @p SubModelFactory template for ModelVoltageSetPointChangeFactory Model
+ */
+class ModelVoltageSetPointChangeFactory : public SubModelFactory {
+ public:
+  /**
+   * @brief default constructor
+   *
+   */
+  ModelVoltageSetPointChangeFactory() { }
+  /**
+   * @brief default destructor
+   *
+   */
+  ~ModelVoltageSetPointChangeFactory() { }
+
+  /**
+   * @brief Model VoltageSetPointChange getter
+   *
+   * @return A pointer to a new instance of Model VoltageSetPointChange
+   */
+  SubModel* create() const;
+
+  /**
+   * @brief Model VoltageSetPointChange destroy
+   */
+  void destroy(SubModel*) const;
+};
+
+/**
+ * class ModelVoltageSetPointChange
+ */
+class ModelVoltageSetPointChange : public ModelCPP::Impl {
+ public:
+  /**
+   * @brief define type of calculated variables
+   *
+   */
+  typedef enum {
+    nbCalculatedVars_ = 0
+  } CalculatedVars_t;
+
+  /**
+   * @brief Default constructor
+   *
+   * Creates a new ModelVoltageSetPointChange instance.
+   */
+  ModelVoltageSetPointChange();
+  /**
+   * @brief Default destructor
+   *
+   * Creates a new ModelVoltageSetPointChange instance.
+   */
+  ~ModelVoltageSetPointChange() { }
```

Comment:
missing virtual

Reply:
virtual not needed

Reply:
virtual is mandatory when you inherit from a parent class

Reply:
Note: in a derived class, if your base class has a virtual destructor, your own destructor is automatically virtual.

https://www.cs.technion.ac.il/users/yechiel/c++-faq/virtual-dtors.html

Reply:
~ModelCPP::Impl is not virtual....

Reply:
I went up to the top and subModel destructor is virtual so we should be ok. Anyway I think it makes easy to think that the virtualness is not well made so I would prefer to add virtual at all levels

Reply:
We can add virtual everywhere but this has nothing to do with my dev and should be done in another issue to have everything harmonized

Reply:
yes